### PR TITLE
Add OAP Server CRDs

### PR DESCRIPTION
**Pull request by hanahmily**
Tips for reviewers, we should pay attention to `oapserver_types.go` because most of the files are generated by kubebuilder.

Signed-off-by: Gao Hongtao <hanahmily@gmail.com>

### CODE REVIEW

File: api/v1alpha1/oapserver_types.go [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// OAPServerSpec defines the desired state of OAPServer
+type OAPServerSpec struct {
+	// Version of OAP.
+	Version string `json:"version"`
```
Comment (withlin) [inline api/v1alpha1/oapserver_types.go]: What do you think of using `Template corev1.PodTemplateSpec `json:" template "` `?

Comment (hanahmily) [reply api/v1alpha1/oapserver_types.go]: It makes much sense to me. If we use PodTemplateSpec, fields following `Config`(NodeSelector, Affinity....) can be merged into it.

File: api/v1alpha1/oapserver_types.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	// Image is the OAP Server Docker image to deploy.
+	Image string `json:"image,omitempty"`
+	// Count is the number of OAP servers
+	Instances int `json:"instances,imitempty"`
```
Comment (wayilau) [inline api/v1alpha1/oapserver_types.go]: We need add failedlimits for our  oap instances. then choose the releated policy when  limits reached. I think.

Comment (hanahmily) [reply api/v1alpha1/oapserver_types.go]: I don't think it's a common option for the user, if they want it, `podTemplate` is a good choice for them to set up `restartPolicy` and `backoffLimit`

**Pull request merged**